### ISSUE TITLE
`before_batch` not supported by `SimpleExtension`

### ISSUE THREAD

**Issue by pbrakel** [mila-udem/blocks#1125]
This happens when I try to set `before_batch` for an extension that inherits from `SimpleExtension`:

```
In [4]: from blocks.extensions import Timing

In [5]: test = Timing(before_batch=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-8324cab9b168> in <module>()
----> 1 test = Timing(before_batch=True)

/u/brakelp/blocks/blocks/extensions/__init__.pyc in __init__(self, prefix, **kwargs)
    597         kwargs.setdefault('before_first_epoch', True)
    598         kwargs.setdefault('after_epoch', True)
--> 599         super(Timing, self).__init__(**kwargs)
    600         self.current = {
    601             level: {'train': 0, 'read_data': 0}

/u/brakelp/blocks/blocks/extensions/__init__.pyc in __init__(self, **kwargs)
    230                 super_kwargs[key] = value
    231         self.set_conditions(**conditions)
--> 232         super(SimpleExtension, self).__init__(**super_kwargs)
    233 
    234     def set_conditions(self, **kwargs):

TypeError: __init__() got an unexpected keyword argument 'before_batch'
```

My guess is that this happens because it doesn't appear in the set of strings `BOOLEAN_TRIGGERS` of the `SimpleExtension` class. If that's all that needs to be fixed, I'll make a PR for it.

**Comment by dmitriy-serdyuk** [comment]
Yep, it was forgotten. It should be an easy fix.

**Comment by dmitriy-serdyuk** [comment]
Fixed by #1126
